volttron/platform/vip/agent/core.py

```diff
@@ -900,8 +900,10 @@ def __init__(self, owner, address=None, identity=None, context=None,
 
         # self._event_queue = gevent.queue.Queue
         self._event_queue = Queue()
+        # if address is a pika.ConnectionParameters then this is a remote connection to this
+        # rabbitmq so we let the identity be the user.
         if isinstance(self.address, pika.ConnectionParameters):
-            self.rmq_user = self.identity
+            self.rmq_user = '.'.join([instance_name, self.identity])
```

Comment:
This is same as the else statement below. Moreover what if instance_name is None?

Reply:
Actually its not the same.  Note the difference is self.instance_name defaults above to volttron1...while this is using the instance_name that is passed into the function.  Also instance_name is only passed in when we are on a remote platform...however this logic is not what I would call foolproof so I will make a change to make this easier to track/debug.